calculator/main.py

```diff
@@ -24,10 +23,19 @@ def perform_operation(calc, operation, params):
     :param calc: A calculator.
     :param operation: String with the operation name. ie: 'add'
     :param params: Tuple containing the list of nums to operate with.
-                   ie: (1, 2, 3, 4.5, -2)
+                  ie: (1, 2, 3, 4.5, -2)
     """
-    pass
-
+    if operation not in calc['operations']:
+        raise InvalidOperation("Given operation is invalid.")
+    for item_in_params in params:
+        print(type(item_in_params))
+        if type(item_in_params) != int and type(item_in_params) != float:
```

Comment:
Might've wanted to just have a list of types you want to check for. e.g. `accepted_param_types = [int, float]`

This way you can check for both and not have to write such a long statement.